packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -0,0 +1,259 @@
+import type { ReactElement } from 'react';
+import styled from 'styled-components';
+import getIn from '../../../utils/getIn';
+import type { BaseTextProps } from '../../Typography/BaseText';
+import BaseText from '../../Typography/BaseText';
+import type { Theme } from '../../BladeProvider';
+import { useTheme } from '../../BladeProvider';
+import type { IconComponent, IconProps, IconSize } from '../../Icons';
+import makeSpace from '../../../utils/makeSpace';
+import type { TypographyPlatforms } from '../../../tokens/global/typography';
+import makeBorderSize from '../../../utils/makeBorderSize';
+import StyledBaseButton from './StyledBaseButton';
+import type { ButtonMinHeight, ButtonTypography } from './buttonTokens';
+import { typography } from './buttonTokens';
+
+export type BaseButtonIcon = undefined | null | IconComponent;
+
+type BaseButtonCommonProps = {
+  variant?: 'primary' | 'secondary' | 'tertiary' | 'link';
+  intent?: 'positive' | 'negative' | 'notice' | 'info' | 'neutral';
+  contrast?: 'low' | 'high';
+  size?: 'large' | 'medium' | 'small' | 'xsmall';
+  iconPosition?: 'left' | 'right';
+  isDisabled?: boolean;
+  isFullWidth?: boolean;
+  onClick?: () => void;
+  type?: 'button' | 'reset' | 'submit';
+};
+
+type BaseButtonWithoutIconProps = BaseButtonCommonProps & {
+  icon?: undefined;
+  children: string;
+};
+
+type BaseButtonWithIconProps = BaseButtonCommonProps & {
+  icon: IconComponent;
+  children?: string;
+};
+
+export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
+
+const StyledBaseText = styled(BaseText)(
```

Comment:
```suggestion
const ButtonText = styled(BaseText)(
```
Better to not confuse with BaseText or StyledBaseText? this name is anyways not conflicting and contextual too